libusb/os/windows_winusb.c

```diff
@@ -2454,29 +2454,17 @@ static int winusbx_open(int sub_api, struct libusb_device_handle *dev_handle)
 {
 	struct winusb_device_priv *priv = usbi_get_device_priv(dev_handle->dev);
 	struct winusb_device_handle_priv *handle_priv = get_winusb_device_handle_priv(dev_handle);
-	HANDLE file_handle;
 	int i;
 
 	CHECK_WINUSBX_AVAILABLE(sub_api);
+	(void)dev_handle;
 
 	// WinUSB requires a separate handle for each interface
 	for (i = 0; i < USB_MAXINTERFACES; i++) {
 		if ((priv->usb_interface[i].path != NULL)
 				&& (priv->usb_interface[i].apib->id == USB_API_WINUSBX)) {
-			file_handle = windows_open(dev_handle, priv->usb_interface[i].path, GENERIC_READ | GENERIC_WRITE);
-			if (file_handle == INVALID_HANDLE_VALUE) {
-				usbi_err(HANDLE_CTX(dev_handle), "could not open device %s (interface %d): %s", priv->usb_interface[i].path, i, windows_error_str(0));
-				switch (GetLastError()) {
-				case ERROR_FILE_NOT_FOUND: // The device was disconnected
-					return LIBUSB_ERROR_NO_DEVICE;
-				case ERROR_ACCESS_DENIED:
-					return LIBUSB_ERROR_ACCESS;
-				default:
-					return LIBUSB_ERROR_IO;
-				}
-			}
 
-			handle_priv->interface_handle[i].dev_handle = file_handle;
+			handle_priv->interface_handle[i].dev_handle = INVALID_HANDLE_VALUE;
```

Comment:
Is this needed? Isn't it initialized or reset somewhere else?

Reply:
I don't see any other place (beside cleaning-up) where `dev_handle` is assigned. If my understanding is correct, `winusbx_open` is responsible for initializing tructures for device being opened including priv structures. It seems to be that before call to this function these structures are uninitialized

Reply:
I see. Maybe we can just set it to INVALID_HANDLE_VALUE regardless of the sub API and path? And replace the comment with something that this is for WINUSB lazy opening?

Reply:
comment - ok, make it unconditional - I'm not sure. It should work if my understanding is correct but I'm not feeling that much confident about it. There are other sub-API that, I admit, never used and I tried to keep my changes limited to WinUSB sub-api. I know that's poor argument, but I would like not to spread changes to far away and break something in the process

Reply:
It is a fair argument. I would think the other sub-abi's will re-initialize this later, but as you, I haven't checked what they do.

Reply:
After looking at more of the code and the bigger picture I realize that is must indeed be limited to the sub-api. Other interfaces can already have been opened and should not be messed with, as noted in the comment before winusbx_open().